web/modules/custom/farm_nfa/src/PlanPermissionsRouteChecker.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Drupal\farm_nfa;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\user\Entity\Role;
+
+/**
+ * Checks access for displaying plan permissions route.
+ */
+class PlanPermissionsRouteChecker implements AccessInterface {
+
+  /**
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   Run access checks for this account.
+   */
+  public function access(AccountInterface $account, $plan) {
+    // Load the entity by ID.
+    $entity = \Drupal::entityTypeManager()->getStorage('plan')->load($plan);
```

Comment:
Please use dependency injection instead

Reply:
Now that you think of it, you should be able to upcast the plan directly in the access route so you receive the entity directly and not the id.

Reply:
See NodePreviewAccessCheck for an example

Reply:
yes good idea @pcambra, will update it accordingly, thanks !